llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7539,7 +7539,7 @@ VectorizationFactor LoopVectorizationPlanner::computeBestVF() {
                         CM.CostKind);
   precomputeCosts(BestPlan, BestFactor.Width, CostCtx);
   assert((BestFactor.Width == LegacyVF.Width ||
-          Legal->hasUncountableEarlyExit() ||
+          BestPlan.hasRegionWithEarlyExit() ||
```

Comment:
Thanks for the update. I've been thinking about the naming a bit, and the region/plan doesn't technically have any early exits any more , as we converted those, but the input had one. I don't really have any good suggestions for the naming. Any ideas for a better terminology to use here?

Alternatively we could just inline the check here, with an explanation.

Reply:
OK, done!